core/src/test/java/edu/wpi/grip/core/sources/HttpSourceTest.java

```diff
@@ -0,0 +1,110 @@
+
+package edu.wpi.grip.core.sources;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.events.UnexpectedThrowableEvent;
+import edu.wpi.grip.core.http.GripServer;
+import edu.wpi.grip.core.util.MockExceptionWitness;
+import edu.wpi.grip.util.Files;
+import edu.wpi.grip.util.GRIPCoreTestModule;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.FileEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.junit.After;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ *
+ */
+public class HttpSourceTest {
+
+    private File logoFile;
+
+    private GripServer server;
+    private GRIPCoreTestModule coreTestModule;
+    private HttpSource source;
+    private CloseableHttpClient postClient;
+
+    @Before
+    public void setup() throws Exception {
+        coreTestModule = new GRIPCoreTestModule();
+        coreTestModule.setUp();
+        final Injector injector = Guice.createInjector(coreTestModule);
```

Comment:
If you really need to use dependency injection, you can do this to get the network dependencies.

```java
Guice.createInjector(coreTestModule, new GRIPNetworkModule());
```